internal/keyaccess/dataintegrity.go

```diff
@@ -0,0 +1,72 @@
+package keyaccess
+
+import (
+	gocrypto "crypto"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/cryptosuite"
+	"github.com/goccy/go-json"
+	"github.com/pkg/errors"
+)
+
+// TODO(gabe) integrate signing for Data Integrity as well in https://github.com/TBD54566975/ssi-service/issues/105
+
+// DataIntegrityKeyAccess represents a key access object for data integrity using the JsonWebSignature2020 suite:
+// https://w3c.github.io/vc-jws-2020/
+type DataIntegrityKeyAccess struct {
+	cryptosuite.JSONWebKeySigner
+	cryptosuite.JSONWebKeyVerifier
+	cryptosuite.CryptoSuite
+}
+
+// NewDataIntegrityKeyAccess creates a new DataIntegrityKeyAccess object from a key id and private key, generating both
+// JSON Web Key Signer and Verifier objects.
+func NewDataIntegrityKeyAccess(kid string, key gocrypto.PrivateKey) (*DataIntegrityKeyAccess, error) {
+	if kid == "" {
+		return nil, errors.New("kid cannot be empty")
+	}
+	if key == nil {
+		return nil, errors.New("key cannot be nil")
+	}
+	publicKeyJWK, privateKeyJWK, err := crypto.PrivateKeyToPrivateKeyJWK(key)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not convert private key to JWK: %s", kid)
+	}
+	signer, err := cryptosuite.NewJSONWebKeySigner(kid, *privateKeyJWK, cryptosuite.AssertionMethod)
```

Comment:
Would AssertionMethod every need to be dynamic?

Reply:
it does need to be dynamic, this would require some refactoring on the SDK side which I intend to cover in OSE-15.
for VCs, this is an assertion method, for VPs this would be authentication method.

hope it's alright to keep it in limbo for now.